### PR TITLE
[SMALLFIX] Some refactoring

### PR DESCRIPTION
**Pull request by peisun1115**

### CODE REVIEW

**Comment by peisun1115** [comment]
Jenkins, test this please

File: core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java [hunk 1]
```diff
@@ -51,6 +51,9 @@
 /**
  * The client talks to a block worker server. It keeps sending keep alive message to the worker
  * server.
+ *
+ * Note: Every client instance is associated with a session which is usually created for each block
+ * stream. So be careful when resue this client for multiple blocks.
```
Comment (jsimsa) [inline core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java]: resue?

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java [hunk 2]
```diff
@@ -131,6 +131,7 @@
 
 import javax.annotation.concurrent.NotThreadSafe;
 
+
```
Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: why extra empty line?

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java [hunk 3]
```diff
@@ -527,8 +530,10 @@ public FileInfo getFileInfo(AlluxioURI path)
    * @param inodePath the {@link LockedInodePath} to get the {@link FileInfo} for
    * @return the {@link FileInfo} for the given inode
    * @throws FileDoesNotExistException if the file does not exist
+   * @throws AccessControlException if the access is not allowed
```
Comment (jsimsa) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: s/if the access is not allowed/if permission denied/ ... for consistency

**Comment by jsimsa** [comment]
left some minor nits

File: core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java (hunk 1 above)
Comment (peisun1115) [inline core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java]: done

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java (hunk 2 above)
Comment (peisun1115) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: Removed.

**Comment by peisun1115** [comment]
@jsimsa Thanks for the review. PTAL.

File: core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java [hunk 4]
```diff
@@ -53,7 +53,7 @@
  * server.
  *
  * Note: Every client instance is associated with a session which is usually created for each block
- * stream. So be careful when resue this client for multiple blocks.
+ * stream. So be careful when reuse this client for multiple blocks.
```
Comment (jsimsa) [inline core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java]: reusing

File: core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java [hunk 5]
```diff
@@ -51,6 +51,9 @@
 /**
  * The client talks to a block worker server. It keeps sending keep alive message to the worker
  * server.
+ *
+ * Note: Every client instance is associated with a session which is usually created for each block
+ * stream. So be careful when reuse this client for multiple blocks.
```
Comment (ifcharming) [inline core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java]: nit: s/reuse/resuing

File: core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java (hunk 4 above)
Comment (peisun1115) [inline core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java]: done

**Comment by jsimsa** [comment]
LGTM

File: core/server/src/main/java/alluxio/master/file/PermissionChecker.java [hunk 6]
```diff
@@ -116,13 +115,36 @@ public void checkPermission(Mode.Bits bits, LockedInodePath inodePath)
...
     checkInodeList(user, groups, bits, inodePath.getUri().getPath(), inodeList, false);
   }
 
   /**
+   * Gets the permission to access inodePath for the current client user.
+   *
+   * @param inodePath the inode path
+   * @return the permission
+   */
+  public Mode.Bits getPermission(LockedInodePath inodePath) {
```
Comment (ifcharming) [inline core/server/src/main/java/alluxio/master/file/PermissionChecker.java]: Would it be more accurate to rename to `getPermissionForCurrentUser` ?

Comment (ifcharming) [inline core/server/src/main/java/alluxio/master/file/PermissionChecker.java]: Can you please add a unit test for this method?

File: core/common/pom.xml [hunk 7]
```diff
@@ -108,6 +108,11 @@
       <artifactId>gmetric4j</artifactId>
       <version>${gmetric4j.version}</version>
     </dependency>
+    <dependency>
```
Comment (ifcharming) [inline core/common/pom.xml]: `alluxio-core-server` already has the `protobuf-java` dependency. Would it make sense to remove the dependency in `alluxio-core-server` since it's added to common?

File: core/server/src/main/java/alluxio/master/file/PermissionChecker.java (hunk 6 above)
Comment (peisun1115) [inline core/server/src/main/java/alluxio/master/file/PermissionChecker.java]: done

Comment (peisun1115) [inline core/server/src/main/java/alluxio/master/file/PermissionChecker.java]: I was trying to keep it consistent with checkPermission. How do you think?

File: core/common/pom.xml (hunk 7 above)
Comment (peisun1115) [inline core/common/pom.xml]: Done.

**Comment by peisun1115** [comment]
@ifcharming i've addressed your comments. PTAL? thanks

File: core/server/src/test/java/alluxio/master/file/PermissionCheckTest.java [hunk 8]
```diff
@@ -34,11 +42,13 @@
 import alluxio.security.authorization.Mode;
 import alluxio.security.authorization.Permission;
 import alluxio.security.group.GroupMappingService;
+import alluxio.util.CommonUtils;
 import alluxio.util.io.PathUtils;
 import alluxio.wire.FileInfo;
 import alluxio.wire.TtlAction;
 
 import com.google.common.collect.Lists;
+import org.apache.hadoop.yarn.webapp.hamlet.HamletSpec;
```
Comment (ifcharming) [inline core/server/src/test/java/alluxio/master/file/PermissionCheckTest.java]: nit: why do you need this `HamletSpec` ?

File: core/server/src/main/java/alluxio/master/file/PermissionChecker.java (hunk 6 above)
Comment (ifcharming) [inline core/server/src/main/java/alluxio/master/file/PermissionChecker.java]: sg

File: core/server/src/main/java/alluxio/master/file/meta/InodeTree.java [hunk 9]
```diff
@@ -62,7 +62,7 @@
  */
 @NotThreadSafe
 // TODO(jiri): Make this class thread-safe.
-public final class InodeTree implements JournalCheckpointStreamable {
+public class InodeTree implements JournalCheckpointStreamable {
```
Comment (ifcharming) [inline core/server/src/main/java/alluxio/master/file/meta/InodeTree.java]: is this change intended?

**Comment by peisun1115** [comment]
@jiri @ifcharming  can one of you merge this for me? thanks

**Comment by jiri** [comment]
Who dares summon me? ☠️
